core/src/main/scala/kafka/utils/DelayedItem.scala

```diff
@@ -33,7 +33,11 @@ class DelayedItem(val delayMs: Long) extends Delayed with Logging {
    * The remaining delay time
    */
   def getDelay(unit: TimeUnit): Long = {
-    unit.convert(max(dueMs - Time.SYSTEM.milliseconds, 0), TimeUnit.MILLISECONDS)
+    unit.convert(getDelayMs, TimeUnit.MILLISECONDS)
```

Comment:
Is this still used after this change?

Reply:
It seems like it's only used in a test. So I suggest we remove it (as well as the Delayed implementation).

Reply:
Sure thing. I'll drop it.

Reply:
@ijuma I have removed getDelay and we no longer implement Delayed. In doing so I also switched the fetcher to a monotonic clock, as our existing implementation is dangerous.